src/CheckedFile.h

```diff
@@ -32,6 +32,20 @@
 #include "Common.h"
 
 namespace e57 {
+   /// Tool class to read buffer efficiently without 
+   /// multiplying copy operations
+   class BufferView {

```

Comment:
This class declaration should be in `CheckedFile.cpp` and forward declared here.